addons/delivery/models/delivery_carrier.py

```diff
@@ -166,6 +166,13 @@ def _onchange_country_ids(self):
         if not self.country_ids:
             self.zip_prefix_ids = [Command.clear()]
 
+    @api.returns('self', lambda value: value.id)
+    def copy(self, default=None):
+        default = dict(default or {})
+        if 'name' not in default:
+            default['name'] = _("%s (copy)", self.name)
```

Comment:
It's clear enough like that, but you could eventually use the `setdefault` method; that would make a lot of sense here, and save a line.